src/main/webapp/js/common/react/tests/NavbarLink.test.js

```diff
@@ -3,19 +3,25 @@ import NavbarLink from '../components/NavbarLink.js';
 import '@testing-library/jest-dom/extend-expect';
 import {render, screen} from '@testing-library/react';
 
+const mockUrl = {
+  href: '/home.html',
+  name: 'Home',
+  requiresAuth: false,
+};
+
 describe('NavbarLink component', () => {
   it('is added to the DOM', () => {
-    const elem = render(<NavbarLink href="/home.html" name="Home" />);
+    const elem = render(<NavbarLink url={mockUrl} />);
     expect(elem.container.firstChild).not.toBeNull();
   });
 
   it('displays the name', () => {
-    render(<NavbarLink href="/home.html" name="Home" />);
+    render(<NavbarLink url={mockUrl} />);
     expect(screen.queryByText('Home')).not.toBeNull();
   });
 
   it('contains the intended href', () => {
-    render(<NavbarLink href="/home.html" name="Home" />);
+    render(<NavbarLink url={mockUrl} />);
     expect(screen.queryByText('Home')).toHaveAttribute('href', '/home.html');
   });
 });
```

Comment:
why isn't this a part of navbar.test.js?

Reply:
This is testing the specific functionality of the `NavbarLink` component compared to the functionality of the entire `Navbar` component. I thought it would be best to put the tests relating to the `NavbarLink` in `NavbarLink.test.js` instead of in `Navbar.test.js` to keep things organized. I realize the confusion might be coming from the fact that these tests are somewhat redundant, as the `Navbar` tests indirectly test this functionality as well, and would fail if this functionality did not work. Is that a sign to remove these tests?